pmdarima/arima/seasonality.py

```diff
@@ -26,10 +28,124 @@
+        where `x` is the input signal, `trend` is the overall trend, `seasonal`
+        is the seasonal component, and `random` is the noisy component. The
+        input signal `x` can be mostly reconstructed by the other three
+        components with a number of points missing equal to `f`.
+
+    Notes
+    -----
+    This function is generally used in conjunction with
+    :func:`pmdarima.utils.visualization.decomposed_plot`,
+    which plots the decomposed components. Also there are two example notebooks
+    within the `examples` folder which mirror existing R examples.
+
+     References
+    ----------
+    .. [1] Example of decompose using both multiplicative and additive types:
+           https://anomaly.io/seasonal-trend-decomposition-in-r/index.html
+
+    .. [2] R documentation for decompose:
+           https://www.rdocumentation.org/packages/stats/versions/3.6.1/topics/decompose
+
+
+    """
+
+    multiplicative = "multiplicative"
+    additive = "additive"
+
+    # Helper function to stay consistent and concise based on 'type'
+    def _decomposer_helper(a, b, type):
+        if type == multiplicative:
+            return a / b
+        else:
+            return a - b
+
+    # Since R's ts class has a frequency as input I think this it acceptable
+    # to ask the user for the frequency.
+    try:
+        assert int(f) > 1 and isinstance(f, int)
+    except (ValueError, AssertionError):
+        raise ValueError("'f' should be a positive integer")
+
+    if filter is None:
+        filter = np.ones((f,)) / f
+
+    # We only accept the values in multiplicative or additive
+    if type not in (multiplicative, additive):
+        err_msg = "'type' can only take values '{}' or '{}'"
+        raise ValueError(err_msg.format(multiplicative, additive))
+
+    # There needs to be at least 2 periods. This is due to the behavior of
+    # convolutions and how they behave with respect to losing endpoints
+    if (x.shape[0] / f) < 2:
+        raise ValueError("time series has no or less than 2 periods")
+
+    # Take half of f for the convolution / sma process.
+    half_f = int(f / 2.)
+    trend = np.convolve(x, filter, mode='valid')
+    trend = trend[:-1]  # we remove the final index.
+    sma_xs = range(half_f, len(trend) + half_f)
+
+    # Remove the effect of the trend on the original signal
+    detrend = _decomposer_helper(x[sma_xs], trend, type)
+
+    # Determine the seasonal effect of the signal
+    m = np.reshape(detrend, (int(detrend.shape[0] / f), f))
+    seasonal = np.nanmean(m, axis=0).tolist()
+    seasonal = np.array(seasonal[half_f:] + seasonal[:half_f])
+    temp = seasonal
+    for i in range(m.shape[0]):
+        seasonal = np.concatenate((seasonal, temp))
+
+    # We buffer the trend component so that it is the same length as the other
+    # outputs. This counters the effects of losing data by the convolution/sma
+    buffer = [np.nan] * half_f
```

Comment:
Another nice way to do this (doesn't matter, because with a one-off like this you won't get a boost. but since @aaronreidsmith commented, I will as well) is simply `np.ones(half_f) * np.nan`

Reply:
I can evaluate to see if it is gives any type of boost. If so, I can definitely make the change.

Reply:
It seems like the preliminary tests were inconclusive so I will just leave as is.